webapp/lib/app/configurator/standard_messages/controller_messages_helper.dart

```diff
@@ -24,9 +37,79 @@ class StandardMessagesManager {
     return lastIndexInGroup + 1;
   }
 
+  Map<String, MessageCategory> fbCategories = {};
   Map<String, MessageCategory> categories = {};
 
+  MessagesDiffData get diffData {
+    Set<String> unsavedCategoryIds = {};
+    Set<String> unsavedGroupIds = {};
+    Set<String> unsavedMessageIds = {};
+
+    List<model.SuggestedReply> editedMessages = [];
+    List<model.SuggestedReply> deletedMessages = [];
+
+    Map<String, model.SuggestedReply> fbMessages = {};
+    Map<String, model.SuggestedReply> messages = {};
+
+    for (var category in categories.values) {
+      category.messages.forEach((message) {
+        messages[message.suggestedReplyId] = message;
+      });
+    }
+
+    for (var category in fbCategories.values) {
+      category.messages.forEach((message) {
+        fbMessages[message.suggestedReplyId] = message;
+      });
+    }
+
+    Set<String> allMessageKeys = Set()..addAll(messages.keys)..addAll(fbMessages.keys);
+    for (var messageId in allMessageKeys) {
+      var fbMessage = fbMessages[messageId];
+      var localMessage = messages[messageId];
```

Comment:
I quite like `local` as a way of referring to the changes shown in the UI - at the risk of becoming a bit verbose, how about we use that everywhere? (e.g. `localCategories`, `localMessages` etc?)

`fb` is too reminiscent of facebook as a shortcut for firebase, so in the same vein as `local`, we could use the prefix `storage`?

Reply:
Makes sense, updated all the methods to local/storage where appropriate